src/lib/serviceRoutinesV2/getEntityAllTypes.cpp

```diff
@@ -84,7 +85,11 @@ std::string getEntityAllTypes
                                  totalTypesP,
                                  noAttrDetail));
   }
-  TIMED_RENDER(answer = response.toJson(ciP->uriParamOptions[OPT_VALUES]));
+
+  if (limit != 0)
+  {
+    TIMED_RENDER(answer = response.toJson(ciP->uriParamOptions[OPT_VALUES]));
+  }
```

Comment:
You shouldn't do the limit=0 check here here. This is probably the cause of the problem in .test described in [previous comment](https://github.com/telefonicaid/fiware-orion/pull/3994#discussion_r783055360).

Instead, you should deal with the limit=0 condition internally at mongoEntityTypes().